eventuals/static-thread-pool.h

```diff
@@ -19,15 +19,36 @@ namespace eventuals {
 ////////////////////////////////////////////////////////////////////////
 
 struct Pinned {
-  Pinned() {}
+  static Pinned ModuloTotalCPUs(unsigned int cpu) {
+    return Pinned(cpu % std::thread::hardware_concurrency());
+  }
+
+  static Pinned Any() {
+    // TODO(benh): pick the least loaded core. This will require
+    // iterating through and checking the sizes of all the "queues"
+    // and then atomically incrementing which ever queue we pick
+    // since we don't want to hold a lock here.
+    return Pinned(0);
+  }
 
-  Pinned(unsigned int core)
-    : core(core) {}
+  static Pinned ExactCPU(unsigned int cpu) {
+    CHECK(cpu < std::thread::hardware_concurrency())
+        << "specified CPU can't be reached";
+    return Pinned(cpu);
+  }
+
+  unsigned int CPU() {
+    return cpu_;
+  }
 
   Pinned(const Pinned& that)
-    : core(that.core) {}
+    : cpu_(that.cpu_) {}
 
-  std::optional<unsigned int> core;
+ private:
+  Pinned(unsigned int cpu)
+    : cpu_(cpu) {}
+
+  unsigned int cpu_;
```

Comment:
Sorry if this got missed in discussion, but let's keep this as a `std::optional` and then we can return a `std::optional` from `cpu()` as well.